From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Charles Kerr <charles@charleskerr.com>
Date: Tue, 9 Aug 2022 12:35:36 -0500
Subject: add electron deps to license credits file

Ensure that licenses for the dependencies introduced by Electron
are included in `LICENSES.chromium.html`

diff --git a/tools/licenses/licenses.py b/tools/licenses/licenses.py
index 07b33f9c3e909216d9282954ebaaef7c56c2dbac..916e6ed19c857c6ed549e98f1b989108e5748776 100755
--- a/tools/licenses/licenses.py
+++ b/tools/licenses/licenses.py
@@ -335,6 +335,31 @@ SPECIAL_CASES = {
         "License": "Apache 2.0",
         "License File": ["//third_party/dawn/third_party/khronos/LICENSE"],
     },
+    os.path.join('third_party', 'electron_node'): {
+        "Name": "Node.js",
+        "URL": "https://github.com/nodejs/node",
+        "License": "MIT",
+        "License File": ["/third_party/electron_node/LICENSE"],
+    },
+    os.path.join('third_party', 'squirrel.mac'): {
+        "Name": "Squirrel",
+        "URL": "https://github.com/Squirrel/Squirrel.Mac",
+        "License": "MIT",
+        "License File": ["/third_party/squirrel.mac/LICENSE"],
+    },
+    os.path.join('third_party', 'squirrel.mac', 'vendor', 'mantle'): {
+        "Name": "Mantle",
+        "URL": "https://github.com/Mantle/Mantle",
+        "License": "MIT",
+        "License File": "/third_party/squirrel.mac/vendor/mantle/LICENSE.md",
+    },
+    os.path.join('third_party', 'squirrel.mac', 'vendor', 'ReactiveObjC'): {
+        "Name": "ReactiveObjC",
+        "URL": "https://github.com/ReactiveCocoa/ReactiveObjC",
+        "License": "MIT",
+        "License File":
+        ["/third_party/squirrel.mac/vendor/ReactiveObjC/LICENSE.md"],
+    },
 }
 
 # These buildtools/third_party directories only contain
